runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -1175,3 +1176,163 @@ J9::ValuePropagation::getParmValues()
 
    TR_ASSERT(parmIterator->atEnd() && parmIndex == numParms, "Bad signature for owning method");
    }
+
+static bool isTakenSideOfAVirtualGuard(TR::Block* block)
+   {
+   TR::TreeTop* lastTT = block->getLastRealTreeTop();
+   for (TR::TreeTop* tt = block->getEntry(); tt != lastTT; tt = tt->getNextTreeTop())
+      {
+      if (tt->getNode()->getNumChildren() > 0
+         && tt->getNode()->getFirstChild()->isTheVirtualCallNodeForAGuardedInlinedCall())
+         return true;
+      }
+   return false;
+   }
+
+static bool skipFinalFieldFoldingInBlock(TR::Block* block)
+   {
+   if (block->isCold() ||
+       TR_FearPointAnalysis::shouldSkipBlock(block) ||
```

Comment:
I'm not a fan of this coupling - it isn't obvious why FearPointAnalysis should be connected to VP. The common concept probably belongs in its own header or something with a more sensible name.

Reply:
Renamed the function to isOSRRelatedBlock and move it to OSRUtils.hpp.